backend/src/cache/server/mutation.go

```diff
@@ -119,14 +119,43 @@ func MutatePodIfCached(req *v1beta1.AdmissionRequest, clientMgr ClientManagerInt
 
 	annotations[ExecutionKey] = executionHashKey
 	labels[CacheIDLabelKey] = ""
-	var maxCacheStalenessInSeconds int64 = -1
-	maxCacheStaleness, exists := annotations[MaxCacheStalenessKey]
+
+	var cacheStalenessInSeconds int64 = -1
+	var userCacheStalenessInSeconds int64 = -1
+	var defaultCacheStalenessInSeconds int64 = -1
+	var maximumCacheStalenessInSeconds int64 = -1
+
+	userCacheStaleness, exists := annotations[MaxCacheStalenessKey]
 	if exists {
-		maxCacheStalenessInSeconds = getMaxCacheStaleness(maxCacheStaleness)
+		log.Printf("userCacheStaleness: %s", userCacheStaleness)
+		userCacheStalenessInSeconds = stalenessToSeconds(userCacheStaleness)
+		log.Printf("userCacheStalenessInSeconds: %d", userCacheStalenessInSeconds)
+	}
+	defaultCacheStaleness, exists := os.LookupEnv("DEFAULT_CACHE_STALENESS")
```

Comment:
If `userCacheStaleness` exists, we shouldn't even care about defaultCacheStaleness, right?

Reply:
Yes, exactly that is happening.  I just read the two environment variables and the Workflow annotation.
Then i calculate from all three values the staleness to use. So if a user/annotation value is set and less than MAXIMUM_CACHE_STALENESS, the user/annotation value is used instead of DEFAULT_CACHE_STALENESS.

```
	var cacheStalenessInSeconds int64 = -1
	var userCacheStalenessInSeconds int64 = -1
	var defaultCacheStalenessInSeconds int64 = -1
	var maximumCacheStalenessInSeconds int64 = -1

	userCacheStaleness, exists := annotations[MaxCacheStalenessKey]
	if exists {
		userCacheStalenessInSeconds = stalenessToSeconds(userCacheStaleness)
	}
	defaultCacheStaleness, exists := os.LookupEnv("DEFAULT_CACHE_STALENESS")
	if exists {
		defaultCacheStalenessInSeconds = stalenessToSeconds(defaultCacheStaleness)
	}
	maximumCacheStaleness, exists := os.LookupEnv("MAXIMUM_CACHE_STALENESS")
	if exists {
		maximumCacheStalenessInSeconds = stalenessToSeconds(maximumCacheStaleness)
	}

	if userCacheStalenessInSeconds < 0 {
		cacheStalenessInSeconds = defaultCacheStalenessInSeconds
	} else {
		cacheStalenessInSeconds = userCacheStalenessInSeconds
	}
	if cacheStalenessInSeconds > maximumCacheStalenessInSeconds {
		cacheStalenessInSeconds = maximumCacheStalenessInSeconds
	}
	log.Printf("cacheStalenessInSeconds: %d", cacheStalenessInSeconds)
```

Reply:
Yes. I was thinking that something like this would be cleaner and free from waste operation. I don't have strong opinion though.

```go
var cacheStalenessInSeconds int64 = -1

cacheStaleness, exists := annotations[MaxCacheStalenessKey]
if !exists {
    cacheStaleness, exists = os.LookupEnv("DEFAULT_CACHE_STALENESS")
} 
if exists {
    cacheStalenessInSeconds = stalenessToSeconds(cacheStaleness)
}
...
if cacheStalenessInSeconds > maximumCacheStalenessInSeconds {
    cacheStalenessInSeconds = maximumCacheStalenessInSeconds
}
```